geode-core/src/test/java/org/apache/geode/security/ClientDataAuthorizationUsingLegacySecurityWithFailoverDUnitTest.java

```diff
@@ -189,14 +189,14 @@ public void dataReaderCanStillOnlyReadAfterFailover() throws Exception {
     });
 
     // Initialize client cache and region. Get the port of the primary connected server.
-    VMProvider server_to_fail = determinePrimaryServer(client);
+    MemberVM server_to_fail = determinePrimaryServer(client);
 
     // Bring down primary server
     server_to_fail.invoke(() -> ClusterStartupRule.getCache().close());
 
     // Confirm failover
-    VMProvider secondaryServer = (server1.equals(server_to_fail)) ? server2 : server1;
-    assertThat(secondaryServer).isEqualTo(determinePrimaryServer(client));
+    MemberVM secondaryServer = (server1.getPort() == server_to_fail.getPort()) ? server2 : server1;
+    await().until(() -> getPrimaryServerPort(client) == secondaryServer.getPort());
```

Comment:
Consider extracting this common bit of code that's used in several tests into a method